File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        int curr = 1;
        for (int i = 1; i < 2 * n; i++){
            if (nums[(i - 1) % n] <= nums[i % n]){
                curr += 1;
            }else{
                curr = 1;
            }
            if (curr == n)
                return true;
        }
        return n == 1;  
    }
}

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        curr = 1
        for i in range(1, 2 * n):
            if nums[(i - 1) % n] <= nums[i % n]:
                curr += 1
            else:
                curr = 1
            if curr == n:
                return True
        return n == 1